### PR TITLE
[AIRFLOW-2178]: Add handling on SLA miss errors

### PR DESCRIPTION
**Pull request by d3cay1-zz**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2178

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
    - Prevent problems with manage_slas from causing issues with scheduling of DAGs by catching exceptions and performing logging, including:
      - Added exception logging for sla_miss_callback handler 
      - Added exception logging for sla miss emailing when issues in sending email occur

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
      - tests/jobs.py:SchedulerJobTest.test_scheduler_sla_miss_callback_exception
      - tests/jobs.py:SchedulerJobTest.test_scheduler_sla_miss_email_exception

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/jobs.py [hunk 1]
```diff
@@ -691,10 +694,15 @@ def manage_slas(self, dag, session=None):
...
-                send_email(
-                    emails,
-                    "[airflow] SLA miss on DAG=" + dag.dag_id,
-                    email_content)
+                try:
+                    send_email(
+                       emails,
+                       "[airflow] SLA miss on DAG=" + dag.dag_id,
+                       email_content)
+                    email_sent = True
```
Comment (Fokko) [inline airflow/jobs.py]: Why set this variable here? It is also set to `True` some lines below.

**Review by Fokko — COMMENTED** [review]
Could you fix the tests and the Flake8 violations?

File: airflow/jobs.py (hunk 1 above)
Comment (d3cay1-zz) [reply airflow/jobs.py]: Right, I missed that.

**Comment by d3cay1-zz** [comment]
The tests should be fixed.  I'm working on the Flake8 violations.

File: airflow/jobs.py [hunk 2]
```diff
@@ -24,7 +24,6 @@
 import psutil
 import signal
 import six
-import socket
```
Comment (d3cay1-zz) [inline airflow/jobs.py:27]: This import was not being used

File: airflow/jobs.py [hunk 3]
```diff
@@ -43,7 +42,6 @@
 from airflow import configuration as conf
 from airflow import executors, models, settings
 from airflow.exceptions import AirflowException
-from airflow.logging_config import configure_logging
```
Comment (d3cay1-zz) [inline airflow/jobs.py:46]: This import was not being used

File: tests/jobs.py [hunk 4]
```diff
@@ -23,7 +23,6 @@
 import os
 import shutil
 import six
-import socket
```
Comment (d3cay1-zz) [inline tests/jobs.py:26]: This import was not being used

File: tests/jobs.py [hunk 5]
```diff
@@ -95,7 +94,7 @@ def test_trigger_controller_dag(self):
         target_dag.clear()
 
         scheduler = SchedulerJob()
-        queue = mock.Mock()
```
Comment (d3cay1-zz) [inline tests/jobs.py:98]: The `Mock` class was already being imported, so I cleaned up these constructor calls

**Comment by d3cay1-zz** [comment]
I have the Flake8 warnings covered and unit tests are working.  The last run succeeded for mysql and sqlite; however, the other DBs had apt-get errors 
```apt-get install failed
$ cat ~/apt-get-update.log
Ign:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty InRelease
Hit:2 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates InRelease
Hit:3 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports InRelease
Hit:4 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty Release
Ign:5 http://dl.google.com/linux/chrome/deb stable InRelease
Ign:6 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 InRelease
Hit:7 http://apt.postgresql.org/pub/repos/apt trusty-pgdg InRelease
Hit:8 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 Release
Hit:9 http://security.ubuntu.com/ubuntu trusty-security InRelease
Hit:11 http://dl.google.com/linux/chrome/deb stable Release
Ign:10 http://toolbelt.heroku.com/ubuntu ./ InRelease
Get:12 http://dl.bintray.com/apache/cassandra 39x InRelease [3,168 B]
Hit:14 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty InRelease
Hit:13 http://toolbelt.heroku.com/ubuntu ./ Release
Hit:15 https://download.docker.com/linux/ubuntu trusty InRelease
Ign:17 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty InRelease
Hit:18 https://dl.hhvm.com/ubuntu trusty InRelease
Hit:21 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty InRelease
Hit:23 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty InRelease
Hit:24 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty InRelease
Hit:25 https://packagecloud.io/computology/apt-backport/ubuntu trusty InRelease
Hit:26 https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease
Hit:27 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease
Hit:28 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty InRelease
Hit:29 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release
Fetched 3,168 B in 1s (2,132 B/s)
Reading package lists...
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
The command "sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install slapd ldap-utils openssh-server mysql-server-5.6 mysql-client-core-5.6 mysql-client-5.6 krb5-user krb5-kdc krb5-admin-server oracle-java8-installer python-selinux" failed and exited with 100 during```

**Comment by d3cay1-zz** [comment]
How do I got about getting this to succeed?

**Comment by d3cay1-zz** [comment]
This is 'good to go' unless there are any concerns.

**Review by Fokko — APPROVED** [review]
LGTM @d3cay1 

Thanks for writing tests

**Pull request closed without merging**